core/server/api/canary/email.js

```diff
@@ -22,12 +23,37 @@ module.exports = {
                 .then((model) => {
                     if (!model) {
                         throw new common.errors.NotFoundError({
-                            message: common.i18n.t('errors.api.email.emailNotFound')
+                            message: common.i18n.t('errors.models.email.emailNotFound')
                         });
                     }
 
                     return model.toJSON(frame.options);
                 });
         }
+    },
+
+    retry: {
+        data: [
+            'id'
+        ],
+        permissions: true,
+        query(frame) {
+            return models.Email.findOne(frame.data, frame.options)
+                .then((model) => {
+                    if (!model) {
+                        throw new common.errors.NotFoundError({
+                            message: common.i18n.t('errors.models.email.emailNotFound')
+                        });
+                    }
+
+                    if (!model.get('status') !== 'failed') {
+                        throw new common.errors.IncorrectUsageError({
+                            message: common.i18n.t('errors.api.email.retryNotAllowed')
+                        });
+                    }
+
+                    return megaService.mega.retryFailedEmail(model);
```

Comment:
Not sure, but we probably need an output serializer for this method?

Reply:
Good point. Added one here as an alias to the `read` serializer https://github.com/TryGhost/Ghost/pull/11410/files#diff-19429d8e8d3f2529d3f3cf9f1b2abf56R8-R10

Reply:
@kevinansfield just checked this one and I think it should've been done a little different (the initial implementation of controller/serializer was off conceptually). Throughout in the "frame" controllers, we pass a clean model to the serializer and do the 'toJSON' call on that level. This is done to have a clear separation on which stage of the "frame" does the transformation, and to be able to call controller methods directly (bypassing serializers). 

I'll push a change to `mega` directly ;)

Reply:
#latcodereviewofretreatcode